engine/src/execution_graph/logic_controllers/PhysicalPlanGenerator.h

```diff
@@ -288,8 +288,9 @@ struct tree_processor {
 					query_graph += (*(child->kernel_unit))["output_b"] >> (*(parent->kernel_unit))["input_b"];
 				} else if ((child_kernel_type == kernel_type::PartitionKernel && parent_kernel_type == kernel_type::MergeStreamKernel)
 									|| (child_kernel_type == kernel_type::PartitionSingleNodeKernel && parent_kernel_type == kernel_type::MergeStreamKernel)) {
-					int num_partitions = this->context->getTotalNodes() * 4; // WSM TODO this is a hardcoded number for now. THis needs to change in the near future
-					auto cache_machine_config =	cache_settings{.type = CacheType::FOR_EACH, .num_partitions = num_partitions};
+					
+					const int MAX_NUM_ORDER_BY_PARTITIONS_PER_NODE = 8; // WSM TODO make this dynamic or configurable. ALso used in generate_distributed_partition_plan function
```

Comment:
Before it was hardcoded for a TOTAL number of partitions. Now i am hardcoding a maximum number of partitions it can use PER NODE. How many partitions it uses will depend on how much data. The reason I am setting a MAX here, is because right now we can only set the number of caches when we first build the graph. Its not dynamic.